src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -485,29 +494,65 @@ static VisualTreeHelper()
 				var childResult = SearchDownForTopMostElementAt(posRelToElement, child.Current!, getVisibility, isChildStale);
 
 				// If we found a stale element in child sub-tree, keep it and stop looking for stale elements
-				if (childResult.stale is { })
+				if (childResult.stale is not null)
 				{
 					stale = childResult.stale;
 					isChildStale = null;
 				}
 
 				// If we found an acceptable element in the child's sub-tree, job is done!
-				if (childResult.element is { })
+				if (childResult.element is not null)
 				{
-					if (isChildStale is { }) // Also indicates that stale is null
+					if (isChildStale is not null) // Also indicates that stale is null
 					{
 						// If we didn't find any stale root in previous children or in the child's sub tree,
 						// we continue to enumerate sibling children to detect a potential stale root.
 
+						TRACE($"+ Searching for stale {isChildStale.Value.Name} branch.");
+
 						while (child.MoveNext())
 						{
-							if (isChildStale(child.Current))
+#if TRACE_HIT_TESTING
+							using var __ = SET_TRACE_SUBJECT(child.Current);
+#endif
+
+							if (isChildStale.Value.Method(child.Current))
 							{
-								stale = SearchDownForStaleBranch(child.Current!, isChildStale);
+								TRACE($"- Is {isChildStale.Value.Name}");
+
+								stale = SearchDownForStaleBranch(child.Current!, isChildStale.Value);
+
+#if TRACE_HIT_TESTING
+								while (child.MoveNext())
+								{
+									using var ___ = SET_TRACE_SUBJECT(child.Current);
```

Comment:
One should be enough?

```suggestion
									using var _ = SET_TRACE_SUBJECT(child.Current);
```

Reply:
@jeromelaban It is not. This is currently considered as a regular variable by the compiler, not a discard.

There is an open proposal somewhere in dotnet/csharplang for that.